src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -617,6 +607,28 @@ private void readOffset() throws ConnectException {
     }
   }
 
+  /**
+   * Read the most recently committed offset from the WAL file.
+   *
+   * @return whether a valid offset was read
+   */
+  private boolean readOffsetFromWAL() {
+    // MemoryWAL is used for testing
+    if (wal instanceof FSWAL) {
+      Pair<Long, String> latestOffsetEntry = ((FSWAL) wal).extractLatestOffsetFromWAL();
+      if (latestOffsetEntry == null) {
+        return false;
+      }
+      log.trace("Last committed offset based on WAL: {}", latestOffsetEntry.getKey());
```

Comment:
Do you mean we should avoid a duplicate call on the next line 623? We can assign the result to a variable to use it in both of these subsequent lines.

Reply:
The point of a log line being in trace is that it's not printed often. But we need to keep in mind that even though the log string might not get fully constructed, any arguments that need to be computed will be computed regardless of log level. Here this might not be very expensive and worth changing but I wanted to highlight that we need to be aware of the cost of trace messages. For example, what's the cost of the `latestOffsetEntry.getKey()` call? For expensive cases you'd need to use `if(isTraceEnabled())`

Reply:
I see, yes that is a good point to keep in mind. We don't want unnecessary computations.